### ISSUE TITLE
Inviting organizations to Space

### ISSUE THREAD

**Issue by techsmyth** [alkem-io/server#4100]
## Description

As a Space admin, I want to invite an organization to join my Space, and as an org admin I want to be able to accept or decline that invitation on behalf of my organization.

## Acceptance criteria

**Space community settings tab**
- [ ] In the member organizations section, add button to invite organizations (also keep the 'add organization button for now: for `platform support`, `beta tester`, and `global admin` roles)
- [ ] Invite flow is similar to the invite user's dialog. Includes a message and the role for the selected organizations: Member or Lead. (Member is always selected and disabled, because there's no way to be Lead without being Member).
- [ ] Invitation should be sent to all organization admins (an org can have multiple admins - not just one)
- [ ] If there are no organization admins at all, invitation should be sent to support@alkem.io, preferably with an indication 
- [ ] Revoke a pending org invitation, following the same behavior as revoking a pending user invitation

**Organization profile settings**
- [ ] Add an "Invitations" tab on the org profile listing pending Space invitations with Accept/Decline 
- [ ] Show message if any

**Other**
- [ ] Invite entry point is settings-only — no equivalent button on the community view
- [ ] ~~Don't say members, but associates~~ — **superseded by ruling R15**: Space-side wording stays "Member organisations" / "Member" / "Lead"; "associate" is the organization-side word and appears nowhere in Space-side copy. (The organization settings tab has since been renamed *Community* → *Associates*.)
- [ ] Adding a search bar for organizations in a Space community tab. Follow search for space members
<img width="1804" height="129" alt="Image" src="https://github.com/user-attachments/assets/a1594673-ba02-4851-bb61-1239344e3228" />

**Notifications**
- [ ] Organization admins receive a notification that the organization is invited to join a Space (in app, push, email - all on)
- [ ] Space admins receive a notification that the organization has accepted or rejected their invitation (in app, push, email - all on)
- [ ] Accepting sends a "welcome" notification to the organization's admins, so the ones who did not accept know that no action is needed
- [ ] Accepting must not trigger a double notification — the "joined" notification should only fire when there was no invitation or application step
- [ ] Add the 2 notifications above to the user notification settings

## Additional Context
Prototype (1-1 UI tested) for the org-side accept/decline UI: https://alkemio-prototype.vercel.app/organization/cityscale/settings/invitations

## Areas that will be affected

- `alkem-io/server` — invite guards, organization opt-out setting, notification events + recipients, ancestor-Space enumeration — [server#6467](https://github.com/alkem-io/server/pull/6467)
- `alkem-io/client-web` — unified invite dialog, Member Organisations list + revoke, organization *Invitations* tab, notification settings row, i18n ×6 — [client-web#10272](https://github.com/alkem-io/client-web/pull/10272)
- `alkem-io/notifications` — email delivery: handlers and templates — [notifications#594](https://github.com/alkem-io/notifications/pull/594)
- `alkem-io/test-suites` — live it-specs + Playwright acceptance walks — [test-suites#632](https://github.com/alkem-io/test-suites/pull/632)
- The `notifications` json of every user is affected, so a verification that nothing is broken/changed from the user notification settings and that only the new values are added is required

Cross-repo spec: `agents-hq/specs/061-organization-space-invitations` (branch `server-4100`). Related: [notifications#356](https://github.com/alkem-io/notifications/issues/356), epic [alkemio#1433](https://github.com/alkem-io/alkemio/issues/1433); fixes [server#4602](https://github.com/alkem-io/server/issues/4602).

## Delivered beyond these acceptance criteria — please confirm

Not requested in this issue; recorded here so the decisions are visible rather than buried in the spec.

1. **Organization opt-out setting.** Organizations get a setting "Allow Spaces to invite this organisation" (default on); inviting an opted-out organization returns the outcome "organisation is not accepting invitations" and creates nothing. *(operator ruling R2)*
2. **Admins only — owners are not notified.** The notifications above go to organization **admins**, exactly as this issue and [notifications#356](https://github.com/alkem-io/notifications/issues/356) specify. Consequence worth a decision: an **owner** who is not also an admin *can* accept the invitation on the organization's behalf (they hold the same underlying credential), but receives no notification that there is anything to accept — and an organization with owners but no admins is treated as having none, so its invitation escalates to support@alkem.io. **Should owners be notified too?**

**Comment by larssondenise** [comment]
Scope update from epic refinement (alkem-io/alkemio#1433):
- This ticket's scope now also includes the **org-side accept/decline UI** — an "Invitations" tab on the org profile listing pending Space invitations with Accept/Decline, mirroring the design for #7207's invite dialog. No separate client-web ticket needed, keeping it here.
- **Revoke**: revoking a pending org invitation should follow the same behavior/logic as revoking a pending user invitation — no new design needed, reuse existing pattern.

**Comment by larssondenise** [comment]
Further refinement (alkem-io/alkemio#1433): accept/decline is **not** gated on a dedicated "org admin" role — it's tied to whoever has access to the org's profile settings generally. Confirm this aligns with the RoleSet work in #4830 before implementing.

Working prototype for the accept/decline UI: https://alkemio-prototype.vercel.app/organization/cityscale/settings/invitations

**Comment by ccanos** [comment]
- The objective of this task is to be able to invite organizations to join space communities. The current implementation allows space admins to just add organization to the community, forcing the orgs, we need to implement something like what we have for users, a space can invite a user, the user gets a notification and they can accept it or reject it.
- All of the invitation process is a state machine, (I think it's based on XState but confirm), I want to reuse as much as possible of the current implementation for invitations for users. In fact, spaces, users and organizations are all actors, so we may need to refactor a bit the invitations system to work with actors.
- We are going to maintain the funcionality of "just add" organizations for `platform support`, `beta tester`, and `global admin` roles. In the case of those user roles, they'll have both options, invite an organization or just add them to their space (server and client authorization validation of this).
- We have a prototype, it's running on port http://localhost:5173/, you can see the Space settings in http://localhost:5173/space/innovation-lab/settings/community at the bottom "Member Organizations". the members, and the pending invitations. It's just gidelines, I want to have something more similar to what we have for users. Also the new styles have colors and stuff, forget that, for now implement everything with the current styles and components.
- There's also the Organizations Settings part in here: http://localhost:5173/organization/green-future-labs/settings/invitations where you can see a new tab in the settings to show the pending invitations. Same 5 states as user invitations, see InvitationLifecycleState. try to reuse as much as possible of the user invitation system, they are all actors.
- The organizations don't have the option to apply to join a space community, only can join by invitation or be just added. (so the implementation is not exactly as for users).
- Organizations can have a role in a community, like users, but in the prototype you'll see that in the prototype they can be admin/lead/member. that's wrong. Organizations can only be Leads or Members of a space, invitations can have the "Lead" role attached so when the organization admin accepts the invitation, the organization becomes inmmediatelly a member and a lead.
- Invites come also with have a message, like for users.
- I'd like to reuse as much components as possible and make the UX as similar as "invite users" as possible, I know the Invite user dialog is only for users, but maybe we can refactor it or make something similar for organizations.
- Notifications:
    - org admin: on received invitation to a space. Not sure if the organization admins can receive a notification directed to an organization actor, but this notification is desirable.
    - space admin: on organization accepted invitation
    - space admin: on organization rejected invitation

**Comment by ccanos** [comment]
Workspace spec linked (forward flow, Constitution §11): `specs/061-organization-space-invitations` on the Alkemio workspace repo (`agents-hq`), branch `server-4100` across server / notifications / client-web / test-suites. Cross-reference tag for PRs: `workspace#061-organization-space-invitations`.

Scope per the operator rulings of 2026-09-03: space → organization invitations only (client-web#7268 associates direction excluded), org opt-out setting, org admin/owner notifications + Invitations tab, inviter accepted/declined notifications, unified invite dialog; also fixes #4602.

**Comment by ccanos** [comment]
The "Accepting must not trigger a double notification" AC is implemented for **both** invitations and applications (ruling R35), as the product email asks — *"the 'joined' notification should only fire when there was no invitation **or application** step"*.

For invitations the dedicated accepted/declined notification takes the suppressed one's place. For applications nothing does: `NotificationEvent` has `SPACE_ADMIN_COMMUNITY_APPLICATION` (fired on **submission**) and no application-**approved** event. So approving an application currently notifies nobody but the approver, where on `develop` every co-admin got email, in-app and push.

That gap is accepted here and tracked separately as alkem-io/server#6476 (sub-issue of epic alkem-io/alkemio#1433), which also carries the open question of which settings row should govern the new event. It needs a Release NN risk line for whichever release carries this branch.

**Comment by Comoque1** [comment]
Tested on DEV env:

GA : Global Admin
SpaceAdmin1: Space Admin
SpaceAdmin2: Space Admin
OrgAdmin: Organization Admin
OrgOwner:  Organization Owner
 
 
Scenario 1: Happy path
- GA Creates ORG1
- GA assigns OrgAdmin and OrgOwner
- SpaceAdmin1 creates SPACE1
- SpaceAdmin1 assigns SpaceAdmin2 as admin SPACE1
 
- SpaceAdmin1 invites ORG1 to join the community as Member
- OrgAdmin gets notification
- OrgOwner does NOT notification
- OrgOwner can still accept the invitation (Has admin privileges but no notification)
- OrgAdmin accepts the invitation
- SpaceAdmin1 and SpaceAdmin2 receive a notification of ORG1 joined SPACE1
 
 
Scenario 2: Invitation rejection
- GA Creates ORG2
- GA assigns OrgAdmin and OrgOwner
- SpaceAdmin1 creates SPACE2
 
- SpaceAdmin1 invites ORG2 to join the community as Member
- OrgAdmin gets notification
- OrgAdmin rejects the invitation
- SpaceAdmin1 receive a notification of ORG2 rejected SPACE1
 
 
Scenario 3: No admins
- GA Creates ORG3
- GA assigns OrgOwner, but NO organization admin
- SpaceAdmin1 creates SPACE3
 
- SpaceAdmin1 invites ORG3 to join the community as Member
- support@alkem.io gets an email
- Only global admin can accept the invitation, or assign an Org administrator
 
 
Scenario 4: Re Invite - THIS  scenario is partially correct. old invitation must first be manually archived in order to get final state. and the follow up invitations are successful.
- GA Creates ORG4
- GA assigns OrgAdmin and OrgOwner
- SpaceAdmin1 creates SPACE4
 
- SpaceAdmin1 invites ORG4 to join the community as Member
- OrgAdmin gets notification
- OrgAdmin rejects the invitation
- SpaceAdmin1 receive a notification of ORG4 rejected SPACE1
 
- SpaceAdmin1 invites again ORG4 to join the community as Member
- OrgAdmin gets notification
- OrgAdmin rejects the invitation
- SpaceAdmin1 receive a notification of ORG4 rejected SPACE1
- ...again and again, shouldn’t block anything (invites are archived??)
 
 
Scenario 5: Not accepting invitations
- GA Creates ORG5
- GA assigns OrgAdmin and OrgOwner
- SpaceAdmin1 creates SPACE5
 
- OrgAdmin disables Settings > allow invitations to join spaces
 
- SpaceAdmin1 invites ORG..X, and ORG5 to join the community as Member
- SpaceAdmin1 gets a dialog saying that ORG5 couldn’t be invited, but the rest are invited correctly
- OrgAdmin does NOT get any notification
- OrgAdmin does not see any invite
 
 
Scenario 6: Global admin still able to add an organization to a community of a space
- GA adds an ORG2 to SPACE2
- OrgAdmin doesn’t get a notification
- SpaceAdmin1 doesn’t get a notification
